src/core/server/saved_objects/migrationsv2/types.ts

```diff
@@ -370,7 +370,7 @@ export interface LegacyDeleteState extends LegacyBaseState {
   readonly controlState: 'LEGACY_DELETE';
 }
 
-export type State =
+export type State = Readonly<
```

Comment:
note: It's not `RecursiveReadonly<...>` but we can add it to make sure State is not mutated